### PR TITLE
[fastcdr] Update to 1.0.25

### PR DESCRIPTION
**Pull request by ex-purple**
**Describe the pull request**

- #### What does your PR fix?
  Update fastcdr to 1.0.25

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by ex-purple** [comment]
@microsoft-github-policy-service agree

**Comment by ex-purple** [comment]
Sorry, can someone help me?
I have no idea what is the problem here

File: ports/fastcdr/portfile.cmake [hunk 1]
```diff
@@ -1,17 +1,15 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO eProsima/Fast-CDR
-    REF 065d49248bd4afbae670836ee1f1c718b9760dde # v1.0.15
-    SHA512 1e011f1848abace94299368a5150f9f7513a676ccdc2b2247cebcb098f7b397e9bd20f5663bc35ea9921b1c91654af39e19b867b73c38bdc5612e0e2b926743a
+    REF v1.0.25
+    SHA512 6a20e8ba61fb516fce35bb27f02db8fb45f287bfd49574dcb39e149e6edde0b4d0598f00399a4d0557ca1f6133d96a6701d6475cbd9039879cddb5089b0ef447
     HEAD_REF master
     PATCHES install-cmake.patch
```
Comment (dg0yt) [inline ports/fastcdr/portfile.cmake:7]: Maybe this patch can be dropped now? The `if` seems to no longer guard `install()`, only `set()`.

**Review by dg0yt — COMMENTED** [review]
I think you want to change this line:
https://github.com/eProsima/Fast-CDR/blob/d6069f345be2e5b34534815a82c3ecfaecb37406/src/cpp/CMakeLists.txt#L198
to:
~~~cmake
if(PDB_FILE AND BUILD_SHARED_LIBS}
~~~

File: ports/fastcdr/portfile.cmake (hunk 1 above)
Comment (ex-purple) [reply ports/fastcdr/portfile.cmake]: 1. Yes, the nstall-cmake.patch is no longer needed.

2. Unfortunately, the problem with the PDB file still persists.

File: ports/fastcdr/portfile.cmake [hunk 2]
```diff
@@ -1,17 +1,16 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO eProsima/Fast-CDR
-    REF 065d49248bd4afbae670836ee1f1c718b9760dde # v1.0.15
-    SHA512 1e011f1848abace94299368a5150f9f7513a676ccdc2b2247cebcb098f7b397e9bd20f5663bc35ea9921b1c91654af39e19b867b73c38bdc5612e0e2b926743a
+    REF v1.0.25
+    SHA512 6a20e8ba61fb516fce35bb27f02db8fb45f287bfd49574dcb39e149e6edde0b4d0598f00399a4d0557ca1f6133d96a6701d6475cbd9039879cddb5089b0ef447
     HEAD_REF master
-    PATCHES install-cmake.patch
+    PATHES
```
Comment (dg0yt) [inline ports/fastcdr/portfile.cmake:7]: Tiny typo, showstopper:
~~~
CMake Warning at scripts/cmake/vcpkg_from_github.cmake:8 (message):
  vcpkg_from_github was passed extra arguments: PATHES;pdb-file.patch
Call Stack (most recent call first):
  ports/fastcdr/portfile.cmake:1 (vcpkg_from_github)
  scripts/ports.cmake:147 (include)
~~~

Comment (ex-purple) [reply ports/fastcdr/portfile.cmake]: My bad

File: ports/fastcdr/portfile.cmake (hunk 1 above)
Comment (ex-purple) [reply ports/fastcdr/portfile.cmake]: Now it's all good

**Comment by MonicaLiu0311** [comment]
The file contains at least one deprecated function, we recommend the new function if you find it convenient:
`vcpkg_install_cmake` -> `vcpkg_cmake_install` (from port `vcpkg-cmake`)
`vcpkg_configure_cmake` -> `vcpkg_cmake_configure` (Please remove the option PREFER_NINJA) (from port `vcpkg-cmake`)
`vcpkg_fixup_cmake_targets` -> `vcpkg_cmake_config_fixup` (from port `vcpkg-cmake-config`)

In the ports that use the new function, you have to add the corresponding dependencies:
```
{
  "name": "vcpkg-cmake",
  "host": true
},
{
  "name": "vcpkg-cmake-config",
  "host": true
}
```

**Comment by MonicaLiu0311** [comment]
```
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
    file(READ ${CURRENT_PACKAGES_DIR}/include/fastcdr/eProsima_auto_link.h EPROSIMA_AUTO_LINK_H)
    string(REPLACE "#define EPROSIMA_LIB_PREFIX \"lib\"" "#define EPROSIMA_LIB_PREFIX" EPROSIMA_AUTO_LINK_H "${EPROSIMA_AUTO_LINK_H}")
    file(WRITE ${CURRENT_PACKAGES_DIR}/include/fastcdr/eProsima_auto_link.h "${EPROSIMA_AUTO_LINK_H}")
else()
    file(READ ${CURRENT_PACKAGES_DIR}/include/fastcdr/config.h FASTCDR_H)
    string(REPLACE "#define _FASTCDR_CONFIG_H_" "#define _FASTCDR_CONFIG_H_\r\n#define FASTCDR_DYN_LINK" FASTCDR_H "${FASTCDR_H}")
    file(WRITE ${CURRENT_PACKAGES_DIR}/include/fastcdr/config.h "${FASTCDR_H}")
endif()
```
It is recommended to use a more convenient function as follows:
```

if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/fastcdr/eProsima_auto_link.h" "#define EPROSIMA_LIB_PREFIX \"lib\"" "#define EPROSIMA_LIB_PREFIX")
else()
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/fastcdr/config.h" "#define _FASTCDR_CONFIG_H_" "#define _FASTCDR_CONFIG_H_\r\n#define FASTCDR_DYN_LINK")
endif()
```

File: ports/fastcdr/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "fastcdr",
-  "version-semver": "1.0.15",
-  "port-version": 1,
+  "version-semver": "1.0.25",
   "description": "eProsima FastCDR is a C++ library that provides two serialization mechanisms. One is the standard CDR serialization mechanism, while the other is a faster implementation that modifies the standard.",
   "homepage": "https://github.com/eProsima/Fast-CDR"
 }
```
Comment (MonicaLiu0311) [inline ports/fastcdr/vcpkg.json:6]: "homepage": "https://github.com/eProsima/Fast-CDR",
  "license": “Apache-2.0”

A `license` is required.

Comment (ex-purple) [reply ports/fastcdr/vcpkg.json]: License added
Thanks for your help

**Comment by ex-purple** [comment]
- [ ] Add license
- [ ] Add dependencies
- [ ] Use more convenient function

File: ports/fastcdr/portfile.cmake [hunk 4]
```diff
@@ -1,18 +1,15 @@
...
+        pdb-file.patch
 )
 
 vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
-    OPTIONS
-        -DCMAKE_DISABLE_FIND_PACKAGE_GTest=ON
-)
+    SOURCE_PATH ${SOURCE_PATH})
```
Comment (MonicaLiu0311) [inline ports/fastcdr/portfile.cmake:12]: vcpkg_configure_cmake -> vcpkg_cmake_configure (Please remove the option PREFER_NINJA) (from port vcpkg-cmake)

Comment (MonicaLiu0311) [reply ports/fastcdr/portfile.cmake:12]: Thank you very much for the edit, but still old functions, note that they are very similar, but the word positions are different.

Comment (ex-purple) [reply ports/fastcdr/portfile.cmake]: Yes, I didn't notice it.
Fixed now.

**Comment by MonicaLiu0311** [comment]
> vcpkg_install_cmake()

vcpkg_install_cmake -> vcpkg_cmake_install (from port vcpkg-cmake)

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by ex-purple** [comment]
@MonicaLiu0311 what's next?

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 what's next?

It's all over, I will suggest the team to Merge as soon as possible.

**Pull request merged**